### DISCUSSION TITLE
Calibration with NL2SOL not exploring full range of design space

### DISCUSSION THREAD

**Discussion by philippe-manne** [snl-dakota/dakota#138] [Q&A]
Hi. I am using NLSOL for calibration work. Stubbornly, it keeps exploring a very limited range of the design space instead of reaching out towards the lower and upper bounds despite residues being high. I am wondering if my problem is ill defined or not. I don't see why it behaves like this. Activating scaling, convergence, ... or not doesn't help. Switching the order of parameters or removing some doesn't change the phenomenon. To facilitate the investigation, I boiled down my problem to this very simple setup which suffers from the same symptoms. In other words, what is preventing NLSOL from checking parameter values like (beta=16, gamma=23) after the initial sensitivity assessments??? It then results into false convergence... Any insight will be appreciated. Thank you in advance. -- Philippe

variables
  continuous_design = 2
    initial_point    20.   18.
    lower_bounds     15.   15.
    upper_bounds     25.   25.
    descriptors     'beta' 'gamma'

yields:
%eval_id interface           beta          gamma       RESIDUE1       RESIDUE2       RESIDUE3       RESIDUE4 
1            NO_ID             20             18        -31.097       -111.867       -229.416        -226.87 
2            NO_ID    19.99674203    17.99821682        -34.908       -115.406       -229.829       -229.983 
3            NO_ID    19.99962816    17.99979595        -36.339       -114.153       -231.036       -228.648 
4            NO_ID    19.99996282    17.99997959         -31.62       -113.892       -229.653       -227.731 
5            NO_ID    19.99999628    17.99999796        -30.869       -113.024       -229.645       -229.138 
6            NO_ID    19.99999963     17.9999998        -32.005       -113.687       -228.849       -228.536 

[dakota.log](https://github.com/user-attachments/files/16996536/dakota.log)
[rim_opt_nls.dat.txt](https://github.com/user-attachments/files/16996469/rim_opt_nls.dat.txt)
[rim_opt_nls.in.txt](https://github.com/user-attachments/files/16996470/rim_opt_nls.in.txt)
[rim_opt_nls.out.txt](https://github.com/user-attachments/files/16996569/rim_opt_nls.out.txt)

**Comment by jadamstephens** [comment]
NL2SOL is a local, gradient-based optimizer. It proceeds "downhill" to the (local) minimum associated with the basin it begins in. Dakota also has global optimizers and a multi-start strategy that would be a better (but more expensive) job of exploring the design space and identifying the global minimum.

[Here's a listing](https://snl-dakota.github.io/docs/6.20.0/users/usingdakota/studytypes/optimization.html#optimization-usage-guidelines) of Dakota's optimization methods broken down by their characteristics.

**Reply by brunorossin** [reply]
Im having the same the problem. Is it possible to make Dakota try a larger step for the next iteration? I believe that would solve it in my case. Thank you!

**Reply by Ramzy1990** [reply]
> NL2SOL is a local, gradient-based optimizer. It proceeds "downhill" to the (local) minimum associated with the basin it begins in. Dakota also has global optimizers and a multi-start strategy that would be a better (but more expensive) job of exploring the design space and identifying the global minimum.
> 
> [Here's a listing](https://snl-dakota.github.io/docs/6.20.0/users/usingdakota/studytypes/optimization.html#optimization-usage-guidelines) of Dakota's optimization methods broken down by their characteristics.

Apologies for bumping this topic up but a lot of collaborators I work with also had this issue.  One indeed has to know which optimizer to use and note the properties of the optimizer.  In general, if someone would like to explore their design space, then heuristic methods like evolutionary algorithms (e.g., genetic algorithm) are better for such purposes.

Just one comment though;  one probably will not find the global minimum numerically and will be a local minimum at best.  Unless it is a very limited design space with many constraints (typically toy problems or verification problems).
